app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt

```diff
@@ -13,25 +14,39 @@ class NewsFragment @Inject constructor() : WolmoFragment<FragmentNewsBinding, Ne
     @Inject
     internal lateinit var toastFactory: ToastFactory
 
-    private var newsAdapter: NewsAdapter = NewsAdapter(arrayListOf())
+    private var newsAdapter: NewsAdapter = NewsAdapter(arrayListOf(), this)
 
     override fun init() {}
 
     override fun layout() = R.layout.fragment_news
 
+    override fun onResume() {
+        super.onResume()
+        if (newsAdapter.itemCount != 0) presenter.refreshNews()
+    }
+
     override fun setListeners() {
         super.setListeners()
         with(binding) {
             swipeRefresh.setOnRefreshListener {
                 presenter.updateNews(UpdateInvokeMethod.PULL_DOWN)
-                swipeRefresh.isRefreshing = false
             }
             recyclerView.infiniteScroll {
                 presenter.updateNews(UpdateInvokeMethod.SCROLL)
             }
         }
     }
 
+    fun updateLike(newId: Int) = presenter.updateLike(newId)
```

Comment:
Same here, name that presenter's method like an event (sorry for repeating the comment)